### PR TITLE
Issue 6341: Validating operator versions

### LINKED ISSUE #6341: Ensure System Tests are not allowed to run with non-supported operator versions
**Describe the bug**
The system test framework is only compatible with pravega operator versions >= 0.5.0 and bookkeeper operator versions >= 0.1.3. However there is no check to prevent users from running the system tests if the user specifies an unsupported operator version by mistake.

**To Reproduce**
Execute the system tests by specifying pravega operator version < 0.5.0 or bookkeeper operator version < 0.1.3

**Screenshots**
N.A.

**Additional information**
We should introduce a check in the build.gradle to allow the system tests to execute only if a supported operator version has been specified by the user.

### PR DESCRIPTION
**Pull request by SrishT**
Signed-off-by: SrishT <Srishti.Thakkar@emc.com>

**Change log description**  
Verifies that system tests are executed only if the pravega operator version specified by the user >= 0.5.0 and bookkeeper operator version specified by the user >= 0.1.3

**Purpose of the change**  
Fixes #6341 

**What the code does**  
Adds a regex in the `build.gradle` to ensure that only supported operator versions can be used to trigger the system tests.

**How to verify it**  
System tests should fail if pravega operator version < 0.5.0 or bookkeeper operator version < 0.1.3 is specified.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1180,6 +1180,16 @@ project('test:system') {
             systemProperty "imageVersion", "INVALID_IMAGE_VERSION"
         }
 
+        // validating that bookkeeper operator version >= 0.1.3
+        if (System.getProperty("bookkeeperOperatorVersion") ==~ /^(0)\.(1)\.(0|1|2)(?:-.+)/) {
```
Comment (RaulGracia) [inline build.gradle:1184]: Can we have a more informative failure message in case any of these 2 checks fail?

File: build.gradle [hunk 2]
```diff
@@ -1180,6 +1180,16 @@ project('test:system') {
...
+            systemProperty "bookkeeperOperatorVersion", "INVALID_BOOKKEEPER_OPERATOR_VERSION"
```
Comment (pbelgundi) [inline build.gradle:1185]: Where is the INVALID_BK_OPERATION_VERSION variable used? Do we exit the gradle task if BK/Pravega Operator version is invalid?

Comment (pbelgundi) [reply build.gradle:1185]: Please make sure to test these changes for both positive and negative usecases

**Review by pbelgundi — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 3]
```diff
@@ -1179,6 +1190,14 @@ project('test:system') {
         if (System.getProperty("imageVersion") == null) {
             systemProperty "imageVersion", "INVALID_IMAGE_VERSION"
         }
+        // validating that bookkeeper operator version >= 0.1.3
+        if (System.getProperty("bookkeeperOperatorVersion") ==~ /^(0)\.(1)\.(0|1|2)(-.+)*/) {
+            throw new GradleException('Requires Bookkeeper Operator Version >= 0.1.3')
```
Comment (shrids) [inline build.gradle:1195]: Perhaps we can use InvalidUserDataException to convey this.
https://docs.gradle.org/current/javadoc/org/gradle/api/InvalidUserDataException.html

Comment (SrishT) [reply build.gradle]: Done

File: build.gradle (hunk 1 above)
Comment (SrishT) [reply build.gradle]: Corrected

File: build.gradle (hunk 2 above)
Comment (SrishT) [reply build.gradle]: @pbelgundi both positive and negative usecases have been tested

**Review by shrids — APPROVED** [review]

File: build.gradle [hunk 4]
```diff
@@ -1084,16 +1084,22 @@ project('test:system') {
         description 'Used to copy test artifacts to the Kubernetes Cluster.'
         dependsOn 'jar'
 
+        environment "dockerRegistryUrl", System.getProperty("dockerRegistryUrl")
+        environment "imagePrefix", System.getProperty("imagePrefix", "pravega")
         environment "tier2Type", System.getProperty("tier2Type", "nfs")
+        environment "pravegaOperatorChartVersion",System.getProperty("pravegaOperatorChartVersion", "latest")
```
[the remark this answers is not in the document]
Comment (anishakj) [reply build.gradle:1090]: is there a chart version for latest? If not we should set the latest released version of charts

File: test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/K8SequentialExecutor.java [hunk 5]
```diff
@@ -171,8 +171,6 @@ private static String getArgs() {
                 "pravegaOperatorVersion",
                 "bookkeeperOperatorVersion",
                 "zookeeperOperatorVersion",
-                "desiredPravegaCMVersion",
```
Comment (anishakj) [inline test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/K8SequentialExecutor.java:174]: If we are removing configmap version from supported version map, provide support from operators that is removing supported version.

Comment (SrishT) [reply test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/K8SequentialExecutor.java:174]: done

File: build.gradle [hunk 6]
```diff
@@ -1179,6 +1190,14 @@ project('test:system') {
         if (System.getProperty("imageVersion") == null) {
             systemProperty "imageVersion", "INVALID_IMAGE_VERSION"
         }
+        // validating that bookkeeper operator version >= 0.1.3
```
Comment (anishakj) [inline build.gradle:1193]: Change the comment to `0.1.5`

**Review by anishakj — APPROVED** [review]
LGTM

**Review by pbelgundi — APPROVED** [review]
LGTM

**Pull request merged**